lebedigital/raw_data_processing/youngs_modulus_data/mixture_metadata_extraction.py

```diff
@@ -0,0 +1,283 @@
+# Script for metadata-extraction: 
+# give path to raw data and output-path, get yaml-file with the metadata
+# If the raw data (excelfile) has multiple sheets, all of them are processed.
+# based on extraction-code of Erik
+
+
+# Workflow:
+# 1. Installing and importing the necessary libraries
+# 2. Define a function to convert german formatting to english
+# 3. Define a function to check for nan value regardless of float or string
+# 4. Defining a function to extract metadata:
+# 3.a Look for all sheets containing the word "Rezeptur" in the file, load them
+#     and go through them one by one
+# 3.b Find indices of the labels, except for labels called "addition" (Zusatzstoff)
+#     (those get renamed to addition1 and addition2 to ensure unique keys), and
+#     create a dictionary with this information
+# 3.c Check for missing labels
+# 3.d Save the name of the addition into the annotation in case it isn't mentioned
+#     there already
+# 3.e Define a function to ignore empty annotations.
+# 3.f Extract the metadata for each label which is existing. 
+# 3.g Depending on wheter an output-path is given or not, return the metadata
+#     dictionary or create a yaml-file.
+
+#------------------------------------------------------------------------------
+
+from cmath import nan
+import pandas as pd
+import os
+import yaml
+from loguru import logger 
+
+
+# function to convert german formatting to english
+def replace_comma(string, format = 'float'):
+    if '---' in string:
+        string = nan  # maybe None? But this will cause errors when float(string)
+        return string
+    elif format == 'float':
+        string = string.replace(',', '.')
+        return float(string)
+    else:
+        string = string.replace(',', '.')
+        return string
+
+
+# function to check for nan-values independently of the format (str/float)
+def isNaN(num):
```

Comment:
why not use standard functions such as [this](https://numpy.org/doc/stable/reference/generated/numpy.isnan.html)